src/Documentation/implementation/streams_implementation/azure_queue_streams.md

```diff
@@ -53,3 +53,62 @@ hostBuilder
     options.QueueNames = new List<string> { "yourprefix-azurequeueprovider-0" };
   }))
 ```
+
+## Tuning
+
+In a production system can emerge the need of tuning over the default configuration. When tuning there are some factors that should be considered, and it's service specific.
+
+1. First, most of the settings are per queue, so for a large number of streams, the load on each queue can be reduced by configuring more queues.
+2. Since streams process messages in order per stream, the gating factor will be the number of events being sent on a single stream.
+3. A reasonable balance of **cache time** (`StreamPullingAgentOptions.GetQueueMsgsTimerPeriod`) vs **visibility time** (`AzureQueueOptions.MessageVisibilityTimeout`) is that the visibility should be configured **to double** the time messages **are expected** to be in the cache.
+
+### Example
+
+We assume a system with this characteristics:
+
+* 100 streams,
+* 10 queues,
+* Each stream processing 60 messages per minute,
+* Each message takes around 30ms to process,
+* 1 minute worth of messages in cache (cache time).
+
+So we can calculate some parameters of the system:
+
+* Streams/queue: Even balancing of streams across queues would be an ideal 10 streams/queue (100 streams / 10 queues).
+But since streams won't always be evenly balanced over the queues, doubling (or even tripling) the ideal is safer than expecting ideal distribution.
+Hence **20 streams/queue** (10 streams/queue x 2 as safety factor) is probably reasonable.
+
+* Messages/minute: This means each queue will be expected to process up to **1200 messages/minute** (60 messages x 20 streams).
+
+The we can determine the visibility time to use:
```

Comment:
"The we can"  => "We can"?

Reply:
or "Then we can"?  Not sure.. your call :)

Reply:
Fixed, thanks